Content.Client/UserInterface/Systems/BugReport/Windows/BugReportWindow.xaml.cs

```diff
@@ -0,0 +1,156 @@
+    // (E.g. the amount of characters they have typed)
+    private void HandleInputChange()
+    {
+        var titleLen = BugReportTitle.Text.Length;
+        var descriptionLen = BugReportDescription.TextLength;
+
+        var invalidTitleLen = titleLen < MinimumBugReportTitleLength || titleLen > MaximumBugReportTitleLength;
+        var invalidDescriptionLen = descriptionLen < MinimumBugReportDescriptionLength || descriptionLen > MaximumBugReportDescriptionLength;
+
+        TitleCharacterCounter.Text = titleLen + "/" + MaximumBugReportTitleLength;
+        TitleCharacterCounter.FontColorOverride = invalidTitleLen ? Color.Red : Color.Green;
+
+        DescriptionCharacterCounter.Text = descriptionLen + "/" + MaximumBugReportDescriptionLength;
+        DescriptionCharacterCounter.FontColorOverride = invalidDescriptionLen ? Color.Red : Color.Green;
+
+        SubmitButton.Disabled = invalidTitleLen || invalidDescriptionLen;
+
+        PlaceholderCenter.Visible = descriptionLen == 0;
+    }
+
+    // Checks if the bug report window should be enabled for this client.
+    private bool IsEnabled([NotNullWhen(false)] out string? errorMessage)
+    {
+        errorMessage = null;
+
+        if (!EnablePlayerBugReports)
+        {
+            errorMessage = Loc.GetString("bug-report-window-disabled-not-enabled");
+            return false;
+        }
+
+        if (TimeSpan.FromSeconds(MinimumPlaytimeBugReports) > _job.FetchOverallPlaytime())
+        {
+            errorMessage = Loc.GetString("bug-report-window-disabled-playtime");
+            return false;
+        }
+
+        if (amountOfBugReportsSubmitted >= MaximumBugReportsPerRound)
+        {
+            errorMessage = Loc.GetString("bug-report-window-disabled-submissions", ("num", MaximumBugReportsPerRound));
+            return false;
+        }
+
+        var timeSinceLastReport = DateTime.UtcNow - lastBugReportSubmittedTime;
+        var timeBetweenBugReports = TimeSpan.FromSeconds(MinimumTimeBetweenBugReports);
+
+        if (timeSinceLastReport <= timeBetweenBugReports)
+        {
+            var time = timeBetweenBugReports - timeSinceLastReport;
+            errorMessage = Loc.GetString("bug-report-window-disabled-cooldown", ("time", $"{new TimeSpan(time.Days, time.Hours, time.Minutes, time.Seconds)}"));
+            return false;
+        }
+
+        return true;
+    }
+
+    // Update the state of the window to display either the bug report window or an error explaining why you can't submit a report.
+    private void UpdateEnabled()
+    {
+        var isEnabled = IsEnabled(out var errorMessage);
+        DisabledLabel.Text = errorMessage;
+
+        DisabledLabel.Visible = !isEnabled;
+        BugReportContainer.Visible = isEnabled;
+    }
+
+    protected override void FrameUpdate(FrameEventArgs args)
+    {
+        base.FrameUpdate(args);
+        UpdateEnabled();
```

Comment:
Does this really need to be run every frame? Since the window is never disposed maybe it would be better to trigger these changes instead?